check/uncheck all buttons

hpaijmans opened the pull request:
Fix issue #156 

Implement check/uncheck all functionality for sensors menu.

The code between the "Graphs" and "Summaries" menu's is shared and therefore the change is applied in both menus.

amanusk commented:
Thanks for opening the PR!
I'll take a look when I get a chance

amanusk reviewed (commented):
Thanks for this addition.

I think this has some merit on its own, but it would be more useful if there is a select/remove all button per sensor (e.g, all frequencies, all temperature etc). 

As it stands, I don't see too much use for someone to just completely remove all graphs.. 

Do you think you'd like to implement something like this?

hpaijmans commented:
Hi,

> As it stands, I don't see too much use for someone to just completely remove all graphs..

The initial thought process was that on some machines, increasingly so in the last years, there are a lot of cores and sensors where you may only want to monitor a few. This is evident by some of the open issues.
I was running into this problem myself and thought a good way to solve it is to disable/enable all but i can see you may want some more fine grained control.

An example of a setup that I was running is the following:
- Enabled some core & mainboard temp graphs
- enable first and last core freq (2 types of cores scaling differently)
- enable average util only.

The fastest way for me to achieve this was to disable all and toggle those I wanted.

> Do you think you'd like to implement something like this?

I will attempt to implement this.

Though it would mean that there would be 3x as many clicks for my workflow. This would still a big improvement of the current so it would help a lot if it where merged with that adjustment.
Now that I think about it, it would make more sense in the selection layout to have them per sensor group.

amanusk reviewed (changes requested):
Thank you! I think this a is very useful addition

A few comments
Currently, the button name depends on the sensor name, which makes sense, but can create weird artifacts and line skips, as seen in the image below, maybe better to have just Check All /Uncheck All

It would be better if all the select/unselect buttons are all at the bottom, above the apply/cancel buttons, and don't change position with the length of the sensor menu, they can be their own widget, and moving between them will be easier, to select/unselect all sensors

Please apply formating with Black (which I should probably add to the CI :)

![s-tui-fix](https://github.com/user-attachments/assets/623ba4b3-4e36-4e04-a775-7fa899be4c4e)

hpaijmans commented:
Hi,

I have changed the layout as requested, added the buttons into a separate columns widget.
Additionally I have left aligned the buttons to allow better association with the columns instead of the sensor name.

Please check with your setup since with my configuration I don't have the "Power" & "Fan" columns and the other columns are equal length.
Therefore I can't confirm that it is working.

> Please apply formatting with Black (which I should probably add to the CI :)

I don't know what you mean with this, what is "Black"?

amanusk commented:
Looks great!
But I'm getting some errors when scrolling down and getting to the buttons:

![s-tui-fix-2](https://github.com/user-attachments/assets/93a312f9-907d-41c7-a91d-b3b060d15c32)

hpaijmans commented:
Hi @amanusk 

I have been able to reproduce the issue.
It seems that you are using a different urwid library version compared to me.

For me the latest urwid library (`urwid==3.0.2`) the issue is not seen. On the version from you screenshot (`urwid==2.4.6`) I have the same error.

I have added a fix for this error.
I tested with both versions of urwid and did not get the error in either `3.0.2` or `2.4.6`.

amanusk commented:
Great!
I don't mind bumping the urwid version to the latest stable, but it can be done after this PR is merged.
I've added format test to CI on master, so just please rebase, apply formatting and I'll merge

amanusk reviewed (approved):
Thanks! looks great
